packages/session-replay/src/BacktraceSessionRecorder.ts

```diff
@@ -0,0 +1,276 @@
+     */
+    readonly maskAllInputs?: boolean;
+
+    /**
+     * Mask specific kinds of input.
+     *
+     * Can be an object with the following keys:
+     * * `color`
+     * * `date`
+     * * `'datetime-local'`
+     * * `email`
+     * * `month`
+     * * `number`
+     * * `range`
+     * * `search`
+     * * `tel`
+     * * `text`
+     * * `time`
+     * * `url`
+     * * `week`
+     * * `textarea`
+     * * `select`
+     * * `password`
+     * @default { password: true }
+     */
+    readonly maskInputOptions?: MaskInputOptions;
+
+    /**
+     * Callback to customize input masking.
+     * @param text input text to mask
+     * @param element input HTML element
+     * @returns masked text
+     * @default undefined
+     * @example
+     * // replace text with letter A repeated for the text length if element has class "mask-a"
+     * maskInputFn = (text, element) =>
+     *   element.classList.contains('mask-a')
+     *     ? 'A'.repeat(text.length)
+     *     : text;
+     */
+    readonly maskInputFn?: MaskInputFn;
+
+    /**
+     * Callback to customize text masking.
+     * @param text text to mask
+     * @returns masked text
+     * @default undefined
+     * @example
+     * // replace text with letter A repeated for the text length
+     * maskTextFn = (text) => 'A'.repeat(text.length);
+     */
+    readonly maskTextFn?: MaskTextFn;
+}
+
+export interface BacktraceSessionRecorderOptions {
+    /**
+     * Maximum recorded event count to be sent with the report. Use `false` to disable the limit.
+     * @default 100
+     */
+    readonly maxEventCount?: number | false;
+
+    /**
+     * Maximum timeframe for recorded events to be sent with the report. Use `false` to disable the limit.
+     * @default false
+     */
+    readonly maxTime?: number | false;
+
+    /**
+     * Sampling options. Use those to reduce event count or size.
+     */
+    readonly sampling?: BacktraceSessionRecorderSamplingOptions;
+
+    /**
+     * Privacy options. Use those to remove confidendial data.
+     */
+    readonly privacy?: BacktraceSessionReplayPrivacyOptions;
+
+    /**
+     * Options passed to `rrweb.record` function. Refer to `rrweb` documentation for more information.
+     */
+    readonly advancedOptions?: recordOptions<Event>;
+}
+
+function defaultIfNotFalse<T>(value: T | false, defaultValue?: T) {
+    if (value === false) {
+        return undefined;
+    }
+
+    return value ?? defaultValue;
+}
+
+export class BacktraceSessionRecorder implements BacktraceAttachment {
+    public readonly type = 'dynamic';
+
+    private readonly _maxEventCount?: number;
+    private readonly _maxTime?: number;
+
+    private _previousEvents?: eventWithTime[] = [];
+    private _events?: eventWithTime[] = [];
+
+    private _stop?: () => void;
+
+    constructor(private readonly _options: BacktraceSessionRecorderOptions) {
+        this._events = [];
+        this._maxEventCount = defaultIfNotFalse(_options.maxEventCount, 100);
```

Comment:
isn't 
`_options.maxEventCount ?? 100`
enough?

Reply:
No, we want to give the user the option to disable this limit (e.g. instead of it, just use the time limit option).